pkg/ioutils/lazy_reader_disk.go

```diff
@@ -163,36 +164,38 @@ func (r *diskBackedLazyReaderAt) ensureOverflowToDisk(till int64) {
 		r.pos = r.maxBufferSize
 	}
 
-	if r.overFlowFile == nil {
+	if r.overflowFile == nil {
 		var err error
+		// "" indicates we want to use os.TempDir().
```

Comment:
making it clear why we do `_ = os.RemoveAll(filepath.Join(os.TempDir(), d.Name()))` in `CleanUpTempFiles`